apps/web/pages/api/logo.ts

```diff
@@ -0,0 +1,106 @@
+import fs from "fs";
+import mime from "mime-types";
+import type { NextApiRequest, NextApiResponse } from "next";
+import path from "path";
+import { z } from "zod";
+
+import { LOGO, LOGO_ICON } from "@calcom/lib/constants";
+import logger from "@calcom/lib/logger";
+
+const log = logger.getChildLogger({ prefix: [`[api/logo]`] });
+
+function removePort(url: string) {
+  return url.replace(/:\d+$/, "");
+}
+
+function extractSubdomainAndDomain(url: string): [string, string] | null {
+  try {
+    const parsedUrl = new URL(url);
+    const hostParts = parsedUrl.href.split(".");
```

Comment:
@zomars you had this as `parsedUrl.hostname` This was always empty for me so we'd always get the Cal.com logo no matter what 🤔